### PR TITLE
🍰 2019/kw22/alpha_data_import

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

fixed data import from alpha data:
- include all collections (commented out)
- refactored neo4j import script
- use of .env file for (additional) configurations / configuration overrides
- lots of fiddeling with neo4j cql files and cypher shell

### Issues

- relates https://github.com/Human-Connection/Human-Connection/issues/692

### Todo

- [ ] catalogue all the data to be imported (will not be done in this PR)
- [ ] import the data (will not be done in this PR)

### CODE REVIEW

File: deployment/legacy-migration/maintenance-worker/migration/neo4j/badges.cql [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CALL apoc.load.json('file:/tmp/mongo-export/splits/current-chunk.json') YIELD value as badge
+CALL apoc.load.json("file:${IMPORT_CHUNK_PATH_CQL}") YIELD value as badge
```
Comment (roschaefer) [inline deployment/legacy-migration/maintenance-worker/migration/neo4j/badges.cql]: I don't think you can expand variables like this. It's a `.cql` file, not a bash script.

File: deployment/legacy-migration/maintenance-worker/migration/neo4j/import.sh [hunk 2]
```diff
@@ -1,17 +1,48 @@
...
-do
-  for chunk in /tmp/mongo-export/splits/$collection/*
+# Import collection function defintion
+function import_collection () {
+  for chunk in ${IMPORT_PATH}splits/$1/*
   do
-    mv $chunk /tmp/mongo-export/splits/current-chunk.json
-    echo "Import ${chunk}" && cypher-shell < $SCRIPT_DIRECTORY/$collection.cql
+    mv $chunk ${IMPORT_CHUNK_PATH}
+    NEO4J_COMMAND="$(envsubst '${IMPORT_CHUNK_PATH_CQL}' < $(dirname "$0")/$1.cql)"
```
Comment (roschaefer) [inline deployment/legacy-migration/maintenance-worker/migration/neo4j/import.sh]: Or is `envsubst` the command to replace `$VARIABLE` in arbitrary files?

File: deployment/legacy-migration/maintenance-worker/migration/neo4j/badges.cql (hunk 1 above)
Comment (ulfgebhardt) [reply deployment/legacy-migration/maintenance-worker/migration/neo4j/badges.cql]: You can if you use magic like I do here: https://github.com/Human-Connection/Human-Connection/pull/697/files#diff-c6d49070ce711f563a75e723b1e8682cR14

File: deployment/legacy-migration/maintenance-worker/migration/neo4j/import.sh (hunk 2 above)
Comment (ulfgebhardt) [reply deployment/legacy-migration/maintenance-worker/migration/neo4j/import.sh]: yes it is

**Comment by ulfgebhardt** [comment]
I would like to merge this with commented out imports to start the next step to analyse which parts are already correctly imported and which are missing - if you agree on that

File: deployment/legacy-migration/maintenance-worker/migration/neo4j/import.sh [hunk 3]
```diff
@@ -1,17 +1,48 @@
...
+#import_collection "organizations"
+#import_collection "pages"
+#import_collection "projects"
+#import_collection "settings"
+#import_collection "shouts"
+#import_collection "status"
+#import_collection "systemnotifications"
+#import_collection "users"
+#import_collection "userscandos"
+#import_collection "usersettings"
```
Comment (roschaefer) [inline deployment/legacy-migration/maintenance-worker/migration/neo4j/import.sh]: I cannot approve these changes. I would like to preserve the status quo, which is a lot more than just badges.

**Pull request merged**